hazelcast/src/main/java/com/hazelcast/internal/util/ConcurrencyUtil.java

```diff
@@ -44,6 +45,20 @@ public String toString() {
         }
     };
 
+    // Default executor for async callbacks: ForkJoinPool.commonPool() or a thread-per-task executor when
+    // the common pool does not support parallelism
+    public static final Executor DEFAULT_ASYNC_EXECUTOR;
+
+    static {
+        Executor asyncExecutor;
+        if (ForkJoinPool.getCommonPoolParallelism() > 1) {
+            asyncExecutor = ForkJoinPool.commonPool();
+        } else {
+            asyncExecutor = command -> new Thread(command).start();
+        }
+        DEFAULT_ASYNC_EXECUTOR = asyncExecutor;
```

Comment:
can we have a way of logging for chosen executor for diagnostic purposes?

Reply:
sounds good, but this is initialized in a static context and IIRC we avoid obtaining `ILogger`s to avoid initializing logging with wrong context (eg disregarding properties in `Config` object). wdyt?

Reply:
Maybe inside a `DiagnosticsPlugin`? But not sure about which one or even if it is worth to create a new one.